pkg/provider/azure_loadbalancer_accesscontrol_test.go

```diff
@@ -0,0 +1,2357 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provider
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2022-07-01/network"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/utils/ptr"
+
+	"sigs.k8s.io/cloud-provider-azure/pkg/azureclients/loadbalancerclient/mockloadbalancerclient"
+	"sigs.k8s.io/cloud-provider-azure/pkg/azureclients/securitygroupclient/mocksecuritygroupclient"
+	"sigs.k8s.io/cloud-provider-azure/pkg/consts"
+	"sigs.k8s.io/cloud-provider-azure/pkg/provider/loadbalancer"
+	"sigs.k8s.io/cloud-provider-azure/pkg/provider/loadbalancer/iputil"
+	"sigs.k8s.io/cloud-provider-azure/pkg/provider/loadbalancer/securitygroup"
+	"sigs.k8s.io/cloud-provider-azure/pkg/provider/loadbalancer/testutil"
+	"sigs.k8s.io/cloud-provider-azure/pkg/provider/loadbalancer/testutil/fixture"
+	"sigs.k8s.io/cloud-provider-azure/pkg/retry"
+)
+
+func TestCloud_reconcileSecurityGroup(t *testing.T) {
+	const (
+		EnsureLB    = true
+		ClusterName = "test-cluster"
+	)
+
+	var (
+		fx      = fixture.NewFixture()
+		k8sFx   = fx.Kubernetes()
+		azureFx = fx.Azure()
+	)
+
+	t.Run("internal Load Balancer", func(t *testing.T) {
+		t.Run("noop when no allow list specified", func(t *testing.T) {
```

Comment:
any tests when ILB services are annotated with allowded CIRDs and service tags?

Reply:
Good catch. Added a few more tests for ILB. They should cover its code path.